File: BaixarXMLNFeDLL/Código Fonte do Exemplo de Uso VB.NET Básico/Form1.vb
Public Class Form1

    Private Sub ButCaptcha_Click(sender As Object, e As EventArgs) Handles ButCaptcha.Click
        If (UDownloadXMLNFeDLL.Captcha("Captcha.gif")) Then
            Img1.Load("Captcha.gif")
        Else
            MessageBox.Show(UDownloadXMLNFeDLL.Msg)
        End If
    End Sub

    Private Sub ButLerCaptcha_Click(sender As Object, e As EventArgs) Handles ButLerCaptcha.Click
        If (UDownloadXMLNFeDLL.CaptchaLer(EditLicencaLeitorCaptcha.Text, "Captcha.gif")) Then
            EditCaptcha.Text = UDownloadXMLNFeDLL.CaptchaLerTexto
        Else
            MessageBox.Show(UDownloadXMLNFeDLL.Msg)
        End If
    End Sub

    Private Sub ButBuscarCertificado_Click(sender As Object, e As EventArgs) Handles ButBuscarCertificado.Click
        EditCertificado.Text = UDownloadXMLNFeDLL.GetCertificadoDigital()
    End Sub

    Private Sub ButBaixar2_Click(sender As Object, e As EventArgs) Handles ButBaixar2.Click
        If (EditCertificado.Text.Trim() = "") Then
            If (UDownloadXMLNFeDLL.BaixarXMLNFeSemCert(EditChave.Text, EditCaptcha.Text, Environment.CurrentDirectory & "\" & EditChave.Text & ".xml")) Then
                Web1.Navigate(Environment.CurrentDirectory & "\" & EditChave.Text & ".xml")
            Else
                MessageBox.Show(UDownloadXMLNFeDLL.Msg)
            End If
        Else
            UDownloadXMLNFeDLL.SetCertificadoDigital(EditCertificado.Text.Trim())
            If (UDownloadXMLNFeDLL.BaixarXMLNFeComCert(EditChave.Text, EditCaptcha.Text, Environment.CurrentDirectory & "\" & EditChave.Text & ".xml")) Then
                Web1.Navigate(Environment.CurrentDirectory & "\" & EditChave.Text & ".xml")
            Else
                MessageBox.Show(UDownloadXMLNFeDLL.Msg)
            End If
        End If
    End Sub
End Class

File: BaixarXMLNFeDLL/Código Fonte do Exemplo de Uso C#.NET Básico/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace DownloadXMLNFeExemploBasico
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
                
        private void ButCaptcha_Click(object sender, EventArgs e)
        {
            if (UDownloadXMLNFeDLL.Captcha("Captcha.gif"))
                Img1.ImageLocation = "Captcha.gif";
            else
                MessageBox.Show(UDownloadXMLNFeDLL.Msg());
        }

        private void ButLerCaptcha_Click(object sender, EventArgs e)
        {
            if (UDownloadXMLNFeDLL.CaptchaLer(EditLicencaLeitorCaptcha.Text, "Captcha.gif"))
                EditCaptcha.Text = UDownloadXMLNFeDLL.CaptchaLerTexto();
            else
                MessageBox.Show(UDownloadXMLNFeDLL.Msg());
        }

        private void ButBuscarCertificado_Click(object sender, EventArgs e)
        {
            EditCertificado.Text = UDownloadXMLNFeDLL.GetCertificadoDigital();

            if (EditCertificado.Text.Trim() != "")
                UDownloadXMLNFeDLL.SetCertificadoDigital(EditCertificado.Text);
        }

        private void ButBaixar2_Click(object sender, EventArgs e)
        {
            if (EditCertificado.Text == "")
            {
                if (UDownloadXMLNFeDLL.BaixarXMLNFeSemCert(EditChave.Text, EditCaptcha.Text, EditChave.Text + ".xml"))
                    Web1.Navigate(Environment.CurrentDirectory + "\\" + EditChave.Text + ".xml");
                else
                    MessageBox.Show(UDownloadXMLNFeDLL.Msg());
            }
            else
            {
                if (UDownloadXMLNFeDLL.BaixarXMLNFeComCert(EditChave.Text, EditCaptcha.Text, EditChave.Text + ".xml"))
                    Web1.Navigate(Environment.CurrentDirectory + "\\" + EditChave.Text + ".xml");
                else
                    MessageBox.Show(UDownloadXMLNFeDLL.Msg());
            }
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            //Caso você tenha uma licença de basta descomentar a linha abaixo
            //UDownloadXMLNFeDLL.DLLLicenca("COLOQUE SUA LICENÇA AQUI");
        }
    }
}
